cumulusci/tasks/apex/batch.py

```diff
@@ -32,83 +34,116 @@ def _run_task(self):
-        elif not self.done_for_sure:
+            self.logger.info(repr(failed_batches))
+            raise SalesforceException(
+                f"There were batch errors: {repr(failed_batches)}"
+            )
+        elif not summary["CountsAddUp"]:
             self.logger.info("The final record counts do not add up.")
             self.logger.info("This is probably related to W-1132237")
-            self.logger.info(repr(vals))
+            self.logger.info(repr(summary))
 
         self.logger.info(
-            "%s took %d seconds to process %d batches.",
-            self.batch["ApexClass"]["Name"],
-            self.delta,
-            self.batch["TotalJobItems"],
+            f"{self.options['class_name']} took {summary['ElapsedTime']} seconds to process {summary['TotalJobItems']} batches."
         )
 
-        return self.success
+    def failed_batches(self, batches: Sequence[dict]):
+        failed_batches = []
+        for batch in batches:
+            if batch["NumberOfErrors"]:
+                failed_batches.append(
+                    {
+                        key: value
+                        for key, value in batch.items()
+                        if key
+                        in {
+                            "Id",
+                            "Status",
+                            "ExtendedStatus",
+                            "NumberOfErrors",
+                            "JobItemsProcessed",
+                            "TotalJobItems",
+                        }
+                    }
+                )
+        return failed_batches
 
     def _poll_action(self):
         # get batch status
-        query_results = self.tooling.query(self._batch_query)
 
-        self.batch = query_results["records"][0]
+        if not self.original_created_date:
+            query_results = self.tooling.query(self._batch_query(date_limit=None))
+            self.original_created_date = parse_api_datetime(
+                query_results["records"][0]["CreatedDate"]
+            )
+        else:
+            query_results = self.tooling.query(
+                self._batch_query(date_limit=self.original_created_date)
+            )
+
+        self.batches = query_results["records"]
+        current_batch = self.batches[0]
+
+        summary = self.summarize_batches(self.batches)
         self.logger.info(
-            "%s: %d of %d (%d failures)",
-            self.batch["ApexClass"]["Name"],
-            self.batch["JobItemsProcessed"],
-            self.batch["TotalJobItems"],
-            self.batch["NumberOfErrors"],
+            f"{self.options['class_name']}: "
+            f"Job{'s' if len(summary['Jobs'])>1 else ''}: {summary['Jobs']}"
+            f"{summary['JobItemsProcessed']} of {summary['TotalJobItems']} "
+            f"({summary['NumberOfErrors']} failures)"
         )
 
-        self.poll_complete = not self._poll_again()
+        self.poll_complete = current_batch["Status"] in COMPLETED_STATUSES
```

Comment:
What if the most recent batch is complete but a different one that started since original_created_date is not?

Reply:
This whole branch is supporting a "coding convention" which is not directly supported by the platform, so we have to make some assumptions about how it works. My understanding of the coding convention is that you trigger Job N+1 after Job N is finished. That's what makes them chained. There is no parallel processing at all. That's certainly the case for Abacus.

I guess a parallel job run could be started by a 3rd party (e.g. a schedule) .In that case the safest thing would be to wait for both jobs to finish which would, I guess entail that this line might need to deal with that inconsistent situation.

Reply:
I was imagining a fan pattern where one job starts multiple other jobs rather than everything being sequential. I don't know if that is realistic -- @davidmreed?

Reply:
In theory the code should support it now but testing parallel stuff is always hard and especially so if I don't have a real-world example yet.

https://github.com/SFDO-Tooling/CumulusCI/pull/1490/files#diff-defe95b2261349328fbef8dd3bceeb93R124

Reply:
Fanout is not supported. Batch and queueable chaining is always N->N+1->N+2. However, @prescod's right that a second parallel chain could be invoked by a 3rd party.